File: backend/wypisy/src/main/java/wypisy/example/wypisy/model/ProcessCategory.java
package wypisy.example.wypisy.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.Digits;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Collection;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Slf4j
public class ProcessCategory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotEmpty
    private String name;

    private String description;





    @OneToMany(mappedBy = "category")
    @JsonIgnore
    private Collection<ManufacturingProcess> processes=new ArrayList<>();




}

File: frontend/src/app/models/processCategory.ts
import { DEPARTMENT } from "../ENUMS/DEPARTMENT";

export class processCategory {
    public id:number;
    public name: string;
    public department:string;
    public description: string;


    constructor(
        id:number,
        name: string,
        department:string,
        description: string

    ){
        this.id=id;
        this.name=name;
        this.department=department;
        this.description=description;
    }



}  
